parlai/agents/fid/fid.py

```diff
@@ -318,6 +323,60 @@ def __init__(self, opt: Opt, shared: TShared = None):
         super().__init__(opt, shared=shared)
 
 
+class GoldDocRetrieverFiDAgent(SearchQueryFiDAgent):
+    """
+    Uses the gold retrieved docs (documents shown to crowdsourcing agents).
+
+    This FiD agents has a mock retriever that picks the retrieved docs from the observed
+    example.
+    """
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        opt = deepcopy(opt)
+        opt['rag_retriever_type'] = RetrieverType.OBSERVATION_ECHO_RETRIEVER.value
+        if opt['rag_retriever_query'] != 'full_history':
```

Comment:
@klshuster regarding our previous conversation on possibility of collision between context for document retrieval, I see this as an easy solution to avoid that. I believe forcing the use of full history for this type of agent makes sense, given that the whole retrieval process is recalling some past conversation. Let me know what you think.

Reply:
yep, makes sense to me